sympy/sets/sets.py

```diff
@@ -1281,6 +1331,21 @@ def _measure(self):
             parity *= -1
         return measure
 
+    @property
+    def _kind(self):
+        # below 2 lines are for case where evaluate = False and args contain EmptySet
+        EmptySet = S.EmptySet
+        args = tuple(arg.kind for arg in self.args if arg is not EmptySet)
```

Comment:
Rather than duplicating this code everywhere there could be a helper in the superclass for filtering out the empty set and giving the union of the kinds.